specs/N3GOV-v2.d.ts

```diff
@@ -0,0 +1,244 @@
+/**
+ * Data structures for governance proposals presented to the Nation3 DAO
+ * Version 2.0
+ */
+
+declare enum ProposalKinds {
+  Meta = 'meta',
+  Law = 'law',
+  Expense = 'expense',
+  ParameterChange = 'parameter-change',
+  TreasuryManagement = 'treasury-management',
+  CustodialTreasuryManagement = 'custodial-treasury-management',
+}
+
+/* Shared types */
+
+/**
+ * Ethereum address
+ * @TJS-pattern ^0x[a-fA-F0-9]{40}$
+ */
+type Address = string
+
+declare namespace TransactionOrigins {
+  export type Ethereum = {
+    /**
+     * @default 1
+     */
+    chainId: 1
+    /**
+     * @title Nation3 DAO Agent triggering the transaction
+     * @default 0x336252602b3a8a0be336ed942228305173e8082b
+     */
+    from:
+      | '0x336252602b3a8a0be336ed942228305173e8082b'
+      | '0x7b81e8d4e82796c9b76284fa4d21e57b8b86a06c'
+  }
+}
+
+/**
+ * @title ERC20 transfer
+ */
+type ERC20Transfer = TransactionOrigins.Ethereum & {
+  /**
+   * @title Recipient
+   */
+  recipient: Address
+  /**
+   * @title Token
+   */
+  token: Address
+  /**
+   * @title Amount
+   * @minimum 0
+   */
+  amount: number
```

Comment:
@luisivan Will this be a human-readable number or `uint256`?

Reply:
That'd be human-readable